src/main/kotlin/tech/harmonysoft/oss/gradle/dist/BuildCustomGradleDistributionTask.kt

```diff
@@ -126,7 +153,8 @@ abstract class BuildCustomGradleDistributionTask : DefaultTask() {
     }
 
     private fun getDistributions(): Collection<String> {
-        val childDirectories = extensionsRootDir.listFiles(FileFilter { it.isDirectory })
+        val extensionsRootDir = this.extensionsRootDir.orNull?.asFile
+        val childDirectories = extensionsRootDir?.listFiles(FileFilter { it.isDirectory })
         return if (childDirectories == null || childDirectories.size < 2) {
             project.logger.lifecycle("using a single custom gradle distribution")
```

Comment:
it's worth to verify that there is at least one file to be included into `init.d` in the resulting custom distribution. Otherwise the whole plugin is pointless

Reply:
Is a change that forces the existence of the directory pointed to by `extensionsRootDir` sufficient?

Reply:
sorry, what do you mean? My suggestion is to fail with a descriptive error if the directory doesn't exist

Reply:
I mean this solution: https://github.com/harmonysoft-tech/custom-gradle-dist-gradle-plugin/pull/15#discussion_r1573696585. Gradle will fail when dir `src/main/resources/init.d` doesn't exist.

Reply:
sorry, don't see any check that if `init.d` directory is empty, we fail with a descriptive message. Did I miss something?